xonsh/built_ins.py

```diff
@@ -637,15 +637,41 @@ def run_subproc(cmds, captured=True):
             hist.last_cmd_out = output
     if not prev_is_proxy and hist.last_cmd_rtn > 0 and ENV.get('RAISE_SUBPROC_ERROR'):
         raise CalledProcessError(hist.last_cmd_rtn, aliased_cmd, output=output)
-    if captured:
+    if captured=='stdout':
         return output
+    elif captured=='object':
+        return CompletedProcess(prev_proc, output)
 
 
-def subproc_captured(*cmds):
+class CompletedProcess:
```

Comment:
Well, if the interface is really the same, it does not make a big difference for me. The problem is that now interface isn't the same.

In currently proposed implementation, `CompletedProcess.stdout` and `CompletedProcess.stderr` are (always empty) streams left over `Popen` object, whereas in `subprocess.CompletedProcess` these are strings of read contents from these streams.

I don't care what the underlying implementation is, as long as interface is the same. But having two classes for completed processes, with different interfaces, will lead to confusion.

And one other thing: I have just noticed that stderr is not being captured now when using `?()`. I think it should be (and of course it should be kept separately from stdout, to allow user access them separately).